ArjunaJTA/jta/classes/com/arjuna/ats/jta/common/JTAEnvironmentBean.java

```diff
@@ -94,33 +95,37 @@
     private volatile String xaResourceRecordWrappingPluginClassName;
     private volatile XAResourceRecordWrappingPlugin xaResourceRecordWrappingPlugin;
 
-	private int asyncCommitPoolSize = 10;
+	private volatile int asyncCommitPoolSize = 10;
 
-    private int orphanSafetyInterval = 20000;
+    private volatile int orphanSafetyInterval = 20000;
     
-    private String commitMarkableResourceTableName = "xids";
+    private volatile String commitMarkableResourceTableName = "xids";
 
-	private Map<String, String> commitMarkableResourceTableNameMap = new HashMap<String, String>();
+	private volatile Map<String, String> commitMarkableResourceTableNameMap
+            = Collections.unmodifiableMap(new HashMap<>());
 
-	private List<String> commitMarkableResourceJNDINames = new ArrayList<String>();
+	private volatile List<String> commitMarkableResourceJNDINames
+            = Collections.unmodifiableList(new ArrayList<>());
```

Comment:
1. This looks odd since it creates an empty list which can never be modified.
2. It does not match with how we implement other collections in the bean (xaRecoveryNodes for example) which makes maintenance more difficult.
3. Now every API user that mutates the returned collection has to be re-coded to catch UnsupportedOperationException. [Remark: if we do want immutable collections then we would need to wait until we move to java 9 which does have them].

The technique does not address issue https://issues.redhat.com/browse/WFLY-13633 which I thought was one of the reasons for revising how we configure CMR collections.

The technique has the advantage of avoiding the need to give callers copies of the collection when it hasn't changed. But we would need to quantify whether or not that benefit outweighs the disadvantages.